aea/aea.py

```diff
@@ -101,32 +101,32 @@ def handle(self, envelope: Envelope) -> None:
         :param envelope: the envelope to handle.
         :return: None
         """
-        protocol = self.resources.protocol_registry.fetch_protocol(envelope.protocol_id)
+        protocol = self.resources.protocol_registry.fetch(envelope.protocol_id)
 
         # fetch the handler of the "default" protocol for error handling. TODO: change with the handler of "error" protocol.
         default_handler = self.resources.handler_registry.fetch("default")
 
         if protocol is None:
             if default_handler is not None:
-                default_handler.send_unsupported_protocol(envelope)
+                default_handler.send_unsupported_protocol(envelope)  # type: ignore
```

Comment:
We have to ignore here as the base class 'Handler' does not know about these attributes.

Reply:
I see. Shall we apply `cast`?

something like:
```
from aea.skills.default.handler import DefaultHandler
default_handler = cast(DefaultHandler, default_handler) 
```

Reply:
That would let us to remove the `# type: ignore` statements.

Reply:
Yeah good point. Will do.